<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Lefser finds features that have greatest differences between classes. • lefser</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Lefser finds features that have greatest differences between classes.">
<meta property="og:description" content="lefser">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-90028752-3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-90028752-3');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">lefser</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.12.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/lefser.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/lefser/" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Lefser finds features that have greatest
differences between classes.</h1>
                        <h4 data-toc-skip class="author"><div class="line-block">Asya Khleborodova, Ludwig Geistlinger, and Levi
Waldron<br><small>School of Public Health, City University of New
York</small>
</div></h4>
            
            <h4 data-toc-skip class="date">2024-03-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/lefser/blob/HEAD/vignettes/lefser.Rmd" class="external-link"><code>vignettes/lefser.Rmd</code></a></small>
      <div class="hidden name"><code>lefser.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Lefser is metagenomic biomarker discovery tool that is based on <a href="https://huttenhower.sph.harvard.edu/galaxy/" class="external-link">LEfSe</a> tool and is
published by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3218848/" class="external-link">Huttenhower
et al. 2011</a>. <code>Lefser</code> is the R implementation of the
<code>LEfSe</code> method.</p>
<p>Using statistical analyses, <code>lefser</code> compares microbial
populations of healthy and diseased subjects to discover differencially
expressed microorganisms. <code>Lefser</code> than computes effect size,
which estimates magnitude of differential expression between the
populations for each differentially expressed microorganism. Subclasses
of classes can also be assigned and used within the analysis.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install Bioconductor and the <code>lefser</code> package, run the
following commands.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"lefser"</span><span class="op">)</span></span></code></pre></div>
<p>Then load the <code>lefser</code> package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/lefser" class="external-link">lefser</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview-and-example-use-of-lefser">Overview and example use of <code>lefser</code><a class="anchor" aria-label="anchor" href="#overview-and-example-use-of-lefser"></a>
</h2>
<p>The <code>lefser</code> function can be used with a
<code>SummarizedExperiment</code>.</p>
<p>Load the <code>zeller14</code> example dataset and exclude ‘adenoma’
conditions.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">)</span></span>
<span><span class="va">zeller14</span> <span class="op">&lt;-</span> <span class="va">zeller14</span><span class="op">[</span>, <span class="va">zeller14</span><span class="op">$</span><span class="va">study_condition</span> <span class="op">!=</span> <span class="st">"adenoma"</span><span class="op">]</span></span></code></pre></div>
<p>Note. <code>lefser</code> supports only two-group contrasts.</p>
<p>The <code>colData</code> in the <code>SummarizedExperiment</code>
dataset contains the grouping column <code>study_condition</code> which
includes the ‘control’ and ‘CRC’ groups.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">$</span><span class="va">study_condition</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## control     CRC </span></span>
<span><span class="co">##      66      91</span></span></code></pre>
<p>There can be subclasses in each group condition. In the example
dataset we include <code>age_category</code> as a subclass of
<code>study_condition</code> which includes ‘adults’ and ‘seniors’. This
variable will correspond to the <code>blockCol</code> input
argument.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">$</span><span class="va">age_category</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  adult senior </span></span>
<span><span class="co">##     91     66</span></span></code></pre>
<p>We can create a contingency table for the two categorical
variables.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">$</span><span class="va">age_category</span>, <span class="va">zeller14</span><span class="op">$</span><span class="va">study_condition</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         </span></span>
<span><span class="co">##          control CRC</span></span>
<span><span class="co">##   adult       46  45</span></span>
<span><span class="co">##   senior      20  46</span></span></code></pre>
<p>We can now use the <code>lefser</code> function. It provides results
as a <code>data.frame</code> with the names of selected microorganisms
and their effect size.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lefser.html">lefser</a></span><span class="op">(</span><span class="va">zeller14</span>, groupCol <span class="op">=</span> <span class="st">"study_condition"</span>, blockCol <span class="op">=</span> <span class="st">"age_category"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in lefser(zeller14, groupCol = "study_condition", blockCol =</span></span>
<span><span class="co">## "age_category"): Convert counts to relative abundances with 'relativeAb()'</span></span></code></pre>
<pre><code><span><span class="co">## The outcome variable is specified as 'study_condition' and the reference category is 'control'.</span></span>
<span><span class="co">##  See `?factor` or `?relevel` to change the reference category.</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                                                                                        Names</span></span>
<span><span class="co">## 1                                                k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus</span></span>
<span><span class="co">## 2                                                                                    k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales</span></span>
<span><span class="co">## 3                                                                                                       k__Bacteria|p__Firmicutes|c__Bacilli</span></span>
<span><span class="co">## 4                  k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_sp_5_1_39BFAA</span></span>
<span><span class="co">## 5 k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_sp_5_1_39BFAA|t__GCF_000159975</span></span>
<span><span class="co">## 6                            k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Eubacteriaceae|g__Eubacterium|s__Eubacterium_hallii</span></span>
<span><span class="co">##      scores</span></span>
<span><span class="co">## 1 -6.193941</span></span>
<span><span class="co">## 2 -5.914844</span></span>
<span><span class="co">## 3 -5.903174</span></span>
<span><span class="co">## 4 -5.558376</span></span>
<span><span class="co">## 5 -5.558376</span></span>
<span><span class="co">## 6 -5.507090</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="visualizing-results-with-lefserplot">Visualizing results with <code>lefserPlot</code><a class="anchor" aria-label="anchor" href="#visualizing-results-with-lefserplot"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/lefserPlot.html">lefserPlot</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<p><img src="lefser_files/figure-html/unnamed-chunk-8-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="interoperating-with-phyloseq">Interoperating with phyloseq<a class="anchor" aria-label="anchor" href="#interoperating-with-phyloseq"></a>
</h2>
<p>When using <code>phyloseq</code> objects, we recommend to extract the
data and create a <code>SummarizedExperiment</code> object as
follows:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">phyloseq</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'phyloseq'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:SummarizedExperiment':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     distance</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     sampleNames</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:GenomicRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     distance</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     distance</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>, <span class="st">"study_1457_split_library_seqs_and_mapping.zip"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"phyloseq"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">kostic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/microbio_me_qiime.html" class="external-link">microbio_me_qiime</a></span><span class="op">(</span><span class="va">fp</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Found biom-format file, now parsing it... </span></span>
<span><span class="co">## Done parsing biom... </span></span>
<span><span class="co">## Importing Sample Metdadata from mapping file...</span></span>
<span><span class="co">## Merging the imported objects... </span></span>
<span><span class="co">## Successfully merged, phyloseq-class created. </span></span>
<span><span class="co">##  Returning...</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">unclass</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">kostic</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">kostic</span><span class="op">)</span>, <span class="st">"data.frame"</span><span class="op">)</span></span>
<span><span class="co">## create a SummarizedExperiment object</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span></span>
<span>    assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span><span class="op">)</span>, colData <span class="op">=</span> <span class="va">colData</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 2505 190 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(2505): 304309 469478 ... 206906 298806</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(190): C0333.N.518126 C0333.T.518046 ... 32I9UNA9.518098</span></span>
<span><span class="co">##   BFJMKNMP.518102</span></span>
<span><span class="co">## colData names(71): X.SampleID BarcodeSequence ... HOST_TAXID</span></span>
<span><span class="co">##   Description</span></span></code></pre>
<p>You may also consider using
<code>makeTreeSummarizedExperimentFromPhyloseq</code> from the
<code>mia</code> package (example not shown).</p>
<div class="section level3">
<h3 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h3>
<details><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] phyloseq_1.46.0             lefser_1.12.2              </span></span>
<span><span class="co">##  [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             </span></span>
<span><span class="co">##  [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.6        </span></span>
<span><span class="co">##  [7] IRanges_2.36.0              S4Vectors_0.40.2           </span></span>
<span><span class="co">##  [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      </span></span>
<span><span class="co">## [11] matrixStats_1.2.0           BiocStyle_2.30.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] bitops_1.0-7            permute_0.9-7           sandwich_3.1-0         </span></span>
<span><span class="co">##  [4] rlang_1.1.3             magrittr_2.0.3          multcomp_1.4-25        </span></span>
<span><span class="co">##  [7] ade4_1.7-22             compiler_4.3.2          mgcv_1.9-1             </span></span>
<span><span class="co">## [10] systemfonts_1.0.5       vctrs_0.6.5             reshape2_1.4.4         </span></span>
<span><span class="co">## [13] stringr_1.5.1           pkgconfig_2.0.3         crayon_1.5.2           </span></span>
<span><span class="co">## [16] fastmap_1.1.1           XVector_0.42.0          labeling_0.4.3         </span></span>
<span><span class="co">## [19] utf8_1.2.4              rmarkdown_2.25          ragg_1.2.7             </span></span>
<span><span class="co">## [22] purrr_1.0.2             xfun_0.42               modeltools_0.2-23      </span></span>
<span><span class="co">## [25] zlibbioc_1.48.0         cachem_1.0.8            jsonlite_1.8.8         </span></span>
<span><span class="co">## [28] biomformat_1.30.0       highr_0.10              rhdf5filters_1.14.1    </span></span>
<span><span class="co">## [31] DelayedArray_0.28.0     Rhdf5lib_1.24.2         parallel_4.3.2         </span></span>
<span><span class="co">## [34] cluster_2.1.6           R6_2.5.1                coin_1.4-3             </span></span>
<span><span class="co">## [37] bslib_0.6.1             stringi_1.8.3           jquerylib_0.1.4        </span></span>
<span><span class="co">## [40] Rcpp_1.0.12             bookdown_0.38           iterators_1.0.14       </span></span>
<span><span class="co">## [43] knitr_1.45              zoo_1.8-12              Matrix_1.6-5           </span></span>
<span><span class="co">## [46] splines_4.3.2           igraph_2.0.2            tidyselect_1.2.0       </span></span>
<span><span class="co">## [49] abind_1.4-5             yaml_2.3.8              vegan_2.6-4            </span></span>
<span><span class="co">## [52] codetools_0.2-19        lattice_0.22-5          tibble_3.2.1           </span></span>
<span><span class="co">## [55] plyr_1.8.9              withr_3.0.0             evaluate_0.23          </span></span>
<span><span class="co">## [58] desc_1.4.3              survival_3.5-8          Biostrings_2.70.2      </span></span>
<span><span class="co">## [61] pillar_1.9.0            BiocManager_1.30.22     foreach_1.5.2          </span></span>
<span><span class="co">## [64] generics_0.1.3          RCurl_1.98-1.14         ggplot2_3.5.0          </span></span>
<span><span class="co">## [67] munsell_0.5.0           scales_1.3.0            glue_1.7.0             </span></span>
<span><span class="co">## [70] tools_4.3.2             data.table_1.15.2       fs_1.6.3               </span></span>
<span><span class="co">## [73] mvtnorm_1.2-4           rhdf5_2.46.1            grid_4.3.2             </span></span>
<span><span class="co">## [76] ape_5.7-1               libcoin_1.0-10          colorspace_2.1-0       </span></span>
<span><span class="co">## [79] nlme_3.1-164            GenomeInfoDbData_1.2.11 cli_3.6.2              </span></span>
<span><span class="co">## [82] textshaping_0.3.7       fansi_1.0.6             S4Arrays_1.2.1         </span></span>
<span><span class="co">## [85] dplyr_1.1.4             gtable_0.3.4            sass_0.4.8             </span></span>
<span><span class="co">## [88] digest_0.6.34           SparseArray_1.2.4       TH.data_1.1-2          </span></span>
<span><span class="co">## [91] farver_2.1.1            memoise_2.0.1           htmltools_0.5.7        </span></span>
<span><span class="co">## [94] pkgdown_2.0.7           multtest_2.58.0         lifecycle_1.0.4        </span></span>
<span><span class="co">## [97] MASS_7.3-60.0.1</span></span></code></pre>
</details>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Asya Khleborodova.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
